university-management-system/src/test/java/org/univ/system/UniversityTest.java

```diff
@@ -0,0 +1,44 @@
+package org.univ.system;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class UniversityTest {
+    University university = University.getInstance();
+
+    @Test
+    void shouldParticipantsOfCourse() {
+        Course course = new Course("MAT-123", "Linear algebra", new Professor("Dr. X", Faculty.COMPUTER_SCIENCE, 5));
+        Student s1 = new Student("Zsolt", "654987", Faculty.COMPUTER_SCIENCE);
+        Student s2 = new Student("Bob", "1458723", Faculty.ENGINEERING);
+        course.enrollParticipant(s1);
+        course.enrollParticipant(s2);
+
+        List<Enrollable> result = university.getParticipantsOfCourse(course);
+
+        assertEquals(2, result.size());
+        assertTrue(result.containsAll(List.of(s1, s2)));
+        assertNotSame(course.getEnrolledParticipants(), result);
```

Comment:
can you pls guide me through the logic why you wrote this last assertion and what it stands for?